### DISCUSSION TITLE
Dump optical media (CD+DVD) directly to CHD

### DISCUSSION THREAD

**Discussion by AlexanderRavenheart** [thingsiplay/tochd#5] [Ideas]
This is a porting to the Discussions section of the idea presented in this issue: https://github.com/thingsiplay/tochd/issues/2

One proposal is to use cdrdao (+ toc2cue) for CDs and dd for DVDs, as detailed in this comment: https://github.com/thingsiplay/tochd/issues/2#issuecomment-1244943326
> CDs (PSOne and PS2) can be ripped to bin/cue with cdrdao (+ toc2cue): [Pandorawiki](https://pandorawiki.org/Creating_images_of_PSX_games_using_Linux)
> DVDs (e.g. PS2) should be as straightforward as: dd if=/dev/cdrom of=image_name.iso

Lets discuss the pros and cons of adding such a feature and how to go about it.

**Comment by thingsiplay** [comment]
My thoughts are incomplete and non conclusive.

To me it looks like trouble trying to auto recognize what type of system a CD is, unless there is a simple solution. But as we speak about a single source here, maybe it should be manual type anyway, so the user has to fill in with an option. This could still be useful with integration of the rest.

I still don't know how I would go about the options for the input and output names. In example what output name should it have? There could be some guesses to based on filenames in the original disc, but it's the same problem with guessing what type of system it is. The input paths in the options would need different kind of treatment, as it no longer matches with how the script generates the filenames and paths. The entire logic was not designed for cd direct conversion like this.
